torchvision/transforms/functional.py

```diff
@@ -1375,6 +1375,54 @@ def gaussian_blur(img: Tensor, kernel_size: List[int], sigma: Optional[List[floa
     return output
 
 
+def gaussian_noise(img: Tensor, mean: float, sigma: List[float]) -> Tensor:
+    """Performs Gaussian blurring on the image by given kernel.
+    If the image is torch Tensor, it is expected
+    to have [..., H, W] shape, where ... means an arbitrary number of leading dimensions.
+
+    Args:
+        img (PIL Image or Tensor): Image to be blurred
+        mean (float): Mean of the desired noise corruption.
+        sigma (sequence of floats or float): Gaussian noise standard deviation. Can be a
+            sequence of floats like ``(sigma_x, sigma_y)`` or a single float.
+
+            .. note::
+                In torchscript mode sigma as single float is
+                not supported, use a sequence of length 1: ``[sigma, ]``.
+
+    Returns:
+        PIL Image or Tensor: Gaussian Blurred version of the image.
+    """
+    if not torch.jit.is_scripting() and not torch.jit.is_tracing():
+        _log_api_usage_once(gaussian_noise)
+
+    if sigma is None:
+        raise ValueError("The value of sigma cannot be None.")
+
+    if sigma is not None and not isinstance(sigma, (int, float, list, tuple)):
+        raise TypeError(f"sigma should be either float or sequence of floats. Got {type(sigma)}")
+    if isinstance(sigma, (list, tuple)) and len(sigma) == 1:
+        sigma = [sigma[0], sigma[0]]
+    if isinstance(sigma, (list, tuple)) and len(sigma) == 2:
+        sigma = torch.empty(1).uniform_(sigma[0], sigma[1]).item()
```

Comment:
Functional code should not perform any random calls, this should be done in Transforms

Reply:
Yes will resolve that!

Reply:
yes will resolve this

Reply:
pushed,